vault/auth.go

```diff
@@ -801,7 +801,7 @@ func (c *Core) setupCredentials(ctx context.Context) error {
 		if err != nil {
 			c.logger.Error("failed to create credential entry", "path", entry.Path, "error", err)
 			plug, plugerr := c.pluginCatalog.Get(ctx, entry.Type, consts.PluginTypeCredential, "")
-			if plugerr == nil && plug != nil && !plug.Builtin {
+			if plugerr == nil && (plug == nil || plug != nil && !plug.Builtin) {
```

Comment:
I found this logic very hard to grok, how about splitting it into a few lines to help avoid future bugs?

```suggestion
			found := plugErr == nil && plug != nil
			builtin := found && plug.Builtin
			if !found || !builtin {
```

Reply:
How do you feel about:
```
plug, plugerr := c.pluginCatalog.Get(ctx, entry.Type, consts.PluginTypeCredential, "")
builtin := plug != nil && plug.Builtin
if plugerr == nil && !builtin {
```

Reply:
That looks good to me. While you're there, there's actually another bug, the last argument to `Get` should be `entry.Version`(!)

Reply:
Sounds like some go tests for this will go a long way 😄

Reply:
Re-reading the logic, it doesn't make it terribly clear that one of the reasonably expected failure cases is that the plugin isn't in the catalog at all any more. Maybe at least worth adding a comment for that? i.e. we want to do something special either if the plugin wasn't found at all, or if it was and it's external.